sdk/core/core-http/lib/serviceClient.ts

```diff
@@ -133,16 +131,16 @@ export class ServiceClient {
   /**
    * The ServiceClient constructor
    * @constructor
-   * @param {ServiceClientCredentials} [credentials] The credentials object used for authentication.
+   * @param {TokenCredential} [credentials] The credentials object used for authentication.
    * @param {ServiceClientOptions} [options] The service client options that govern the behavior of the client.
    */
-  constructor(credentials?: ServiceClientCredentials | TokenCredential, options?: ServiceClientOptions) {
+  constructor(credentials?: TokenCredential, options?: ServiceClientOptions) {
     if (!options) {
       options = {};
     }
 
-    if (credentials && !isTokenCredential(credentials) && !credentials.signRequest) {
-      throw new Error("credentials argument needs to implement signRequest method");
+    if (credentials && !isTokenCredential(credentials)) {
+      throw new Error("credentials argument needs to implement TokenCredential interface");
```

Comment:
If given credential is not a `TokenCredential`, then we do an early exit here.
But, a few lines below we have a `if (isTokenCredential(credentials)) / else` block. The `else` part is now dead code right?

Reply:
You're absolutely right, I'll change this logic because we do want to support that `else` case.